src/replication.cpp

```diff
@@ -3021,25 +3169,28 @@ struct redisMaster *replicationAddMaster(char *ip, int port) {
     }
 
     sdsfree(mi->masterhost);
-    mi->masterhost = sdsnew(ip);
-    mi->masterport = port;
+    mi->masterhost = nullptr;
     if (mi->master) {
-        if (FCorrectThread(mi->master))
-            freeClient(mi->master);
-        else
-            freeClientAsync(mi->master);
+        freeClientAsync(mi->master);
```

Comment:
Is this supposed to always free asynchronously or should it free normally if the master is on this thread?